site/cds_rdm/inspire_harvester/reader.py

```diff
@@ -32,26 +33,37 @@ def __init__(
         self._until = until
         self._on_date = on_date
         self._inspire_id = inspire_id
-        #
+
         super().__init__(origin, mode, *args, **kwargs)
 
     def _iter(self, url, *args, **kwargs):
         """Yields HTTP response."""
         headers = {"Accept": "application/json"}
+
+        current_app.logger.info(
+            "Sending the request and iterating through the resulting pages."
+        )
+
         while url:  # Continue until there is no "next" link
+            current_app.logger.info(f"Querying URL: {url}.")
             response = requests.get(url, headers=headers)
             data = response.json()
             if response.status_code == 200:
-                if (
-                    data["hits"]["total"] == 0
-                ):  # TODO make it a warning or info when we have proper logging
-                    raise ReaderError(
+                current_app.logger.info("Request response is successful (200).")
+                if data["hits"]["total"] == 0:
+                    current_app.logger.warning(
                         f"No results found when querying INSPIRE. See URL: {url}."
                     )
 
                 for inspire_record in data["hits"]["hits"]:
+                    current_app.logger.debug(
+                        f"Sending INSPIRE record to transformer. Record details: {inspire_record}."
+                    )
                     yield inspire_record
             else:
+                current_app.logger.error(
+                    f"Request response is not successful. Status code: {response.status_code}."
```

Comment:
```suggestion
                    f"Request to {url} failed. Status code: {response.status_code}."
```
You could add also more details with response error message, however I am hesitant a bit if this log is needed since we raise an error anyways, it will get reported I think (check with Javi on how he reports exceptions)